inject.go

```diff
@@ -69,33 +85,44 @@ func Inject(target interface{}) Option {
 	// 		target.Field(2),  // Baz io.Writer
 	// 	]
 	//
-	// As we iterate through the arguments received by the function, we can
-	// simply copy the value into the corresponding value in the targets list.
+	// As we iterate through the fields of the generated struct, we can simply
+	// copy the value into the corresponding value in the targets list.
 	targets := make([]reflect.Value, 0, t.NumField())
 
 	for i := 0; i < t.NumField(); i++ {
 		f := t.Field(i)
-		// Skip private fields.
-		if f.PkgPath != "" {
+
+		// Skip private fields
+		if f.PkgPath != "" && !f.Anonymous {
```

Comment:
I was thinking about this case:
```golang
        var gave1 *type1
	new1 := func() *type1 {
		gave1 = &type1{}
		return gave1
	}

	var out struct {
		*type1
		T2 *type2 `optional:"true"`
	}

	app := fxtest.New(t,
		Provide(new1),
		Inject(&out),
	)

	defer app.MustStart().MustStop()
	assert.Nil(t, out.type1, "out.type1 should not be set, because it is a private field")
```

Should `out.type1` be set or not? I think it shouldn't, but may be I am missing something.